### DISCUSSION TITLE
QNotify - Apply or overwrite only specific attributes of existing type

### DISCUSSION THREAD

**Discussion by Raymondo97** [quasarframework/quasar#14508] [Ideas / Proposals] [answered]
I'm still a pretty new developer. I noticed in #12877 that `QNotifyOptions` and `QNotifyCreateOptions` is available. How would I be able to use that to define how a new notification is created? Like I mentioned in #14451, I want to avoid creating duplicate buttons.

Additionally--the reason I created a separate discussion for this--my team often adds the same timeout to all of our notifies of the existing `'positive'` notify type. We add a different timeout value to all our `'negative'` notifications. From what I currently understand, if I use the `registerType (typeName, typeOpts)` method to add those timeout values to the existing types, it completely overrides the type. I have no way of applying the existing values of that type, or any preexisting type. 

Let me know if there is any more information you need from me.

**Comment by hawkeye64** [comment, marked as answer]
`QNotifyOptions` and `QNotifyCreateOptions` are TypeScript types only; they do not change Notify behavior or register defaults. They are useful when you want to build a typed options object before passing it to Notify:

```ts
import { Notify, type QNotifyCreateOptions } from 'quasar'

const options: QNotifyCreateOptions = {
  type: 'positive',
  message: 'Saved'
}

Notify.create(options)
```

Your understanding of `registerType()` is correct: registering an existing name replaces that type definition; it does not merge with the built-in definition. To add different timeouts while retaining the built-in appearance, re-register the complete definitions in a boot file:

```ts
import { defineBoot } from '#q-app/wrappers'

export default defineBoot(({ $q }) => {
  $q.notify.registerType('positive', {
    color: 'positive',
    icon: $q.iconSet.type.positive,
    timeout: 2500
  })

  $q.notify.registerType('negative', {
    color: 'negative',
    icon: $q.iconSet.type.negative,
    timeout: 8000
  })
})
```

You could instead register names such as `app-positive` and `app-negative` if you do not want to alter the built-in types.

For actions, Notify currently combines actions from the individual call, global defaults, and the registered type. It does not deduplicate action objects. `ignoreDefaults: true` excludes all global defaults for that notification; there is no option to selectively ignore one default or merge actions by identity. The most predictable approach is to define a shared action in only one place, or wrap `$q.notify()` in an application helper that constructs exactly the defaults and actions your team wants.
